### PR TITLE
DP-22509: Reimplement Allow authors to choose the type of ordered list style they need for a list

### PR DESCRIPTION
**Pull request by arthurbaghdas**
**Description:**
Explain the technical implementation of the work done.

**Jira:** (Skip unless you are MA staff)
DP-22509

**To Test:**
- [ ] Add steps to test this feature

---

[Peer Review Checklist](https://github.com/massgov/openmass/blob/develop/docs/peer_review_checklist.md)

### CODE REVIEW

**Comment by tom-fleming** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>Add a changelog YAML file to this PR</td>
    </tr>
  </tbody>
</table>




<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against a853b57efd77f88c2415594fd5a20e97119265bd
</p>

File: conf/drupal/config/editor.editor.rules_of_court_section.yml [hunk 1]
```diff
@@ -144,6 +144,36 @@ settings:
         -
           label: Checklist
```
Comment (clairesunstudio) [inline conf/drupal/config/editor.editor.rules_of_court_section.yml]: I think we can remove this "checklist" option - this is the same as the disc option stylewise

File: conf/drupal/config/filter.format.basic_html.yml [hunk 2]
```diff
@@ -44,7 +44,7 @@ filters:
...
+      allowed_html: '<br> <p> <h2 id> <h3 id> <h4 id> <h5 id> <h6 id> <ul class="ma__checklist list-style-none list-style-disc list-style-circle list-style-square" start> <ol class="list-style-decimal list-style-decimal-leading-zero list-style-lower-roman list-style-upper-roman list-style-lower-alpha list-style-upper-alpha" start> <cite> <dl> <dt> <dd> <span> <img src alt height width data-entity-type data-entity-uuid data-entity-substitution data-align data-caption> <a hreflang href data-entity-type data-entity-uuid data-entity-substitution> <blockquote cite> <drupal-entity data-entity-type data-entity-substitution data-embed-button alt title data-align data-caption data-entity-embed-display data-entity-embed-display-settings data-view-mode data-entity-uuid data-langcode> <strong> <em> <sub> <sup> <li> <table> <tr> <td rowspan colspan> <th rowspan colspan> <thead> <tbody> <tfoot> <caption>'
```
Comment (clairesunstudio) [inline conf/drupal/config/filter.format.basic_html.yml:47]:
```suggestion
      allowed_html: '<br> <p> <h2 id> <h3 id> <h4 id> <h5 id> <h6 id> <ul class="list-style-none list-style-disc list-style-circle list-style-square" start> <ol class="list-style-decimal list-style-decimal-leading-zero list-style-lower-roman list-style-upper-roman list-style-lower-alpha list-style-upper-alpha" start> <cite> <dl> <dt> <dd> <span> <img src alt height width data-entity-type data-entity-uuid data-entity-substitution data-align data-caption> <a hreflang href data-entity-type data-entity-uuid data-entity-substitution> <blockquote cite> <drupal-entity data-entity-type data-entity-substitution data-embed-button alt title data-align data-caption data-entity-embed-display data-entity-embed-display-settings data-view-mode data-entity-uuid data-langcode> <strong> <em> <sub> <sup> <li> <table> <tr> <td rowspan colspan> <th rowspan colspan> <thead> <tbody> <tfoot> <caption>'
```

**Comment by clairesunstudio** [comment]
The approach looks good! 

I noticed a few things:
- [x] Can we remove the `checklist` option and styles? That one is the same as the `disc` option, let's make that the first option instead.**
- [x] Numbered list and bulleted list don't line up - upon inspection, they have different padding left. Numbered list has `padding-left: 30px` and the bulleted list has `padding-left: 20px`. Let's make them both 20px.**. This is happening in both the front end and the rich text editor.
Current:
Screenshot 2024-08-22 at 3 00 39 PM
Fixed:
Screenshot 2024-08-22 at 3 00 29 PM
- [ ] Not sure if there's a way to fix this, but I think it's confusing that the options are multi-select, and the class names are appended. Ideally, this should be a single option, and only one class should be applied at a time. 
Screenshot 2024-08-22 at 2 57 42 PM
Screenshot 2024-08-22 at 2 57 53 PM
- [x] This can be a little nitpicky, but can we reorder the options so that the upperRoman is before the lower Roman, the same Upper Alpha before Lower Alpha so that are in the general order of hierarchy?

File: conf/drupal/config/editor.editor.basic_html.yml [hunk 3]
```diff
@@ -142,6 +142,36 @@ settings:
         -
```
Comment (clairesunstudio) [inline conf/drupal/config/editor.editor.basic_html.yml:142]:
```suggestion
I think we can remove this "checklist" option - this is the same as the disc option stylewise. Same as elsewhere
```

Comment (arthurbaghdas) [reply conf/drupal/config/editor.editor.basic_html.yml:142]: Removed

**Comment by arthurbaghdas** [comment]
> The approach looks good!
> 
> I noticed a few things:
> 
>     * [x]  Can we remove the `checklist` option and styles? That one is the same as the `disc` option, let's make that the first option instead.**
> 
>     * [x]  Numbered list and bulleted list don't line up - upon inspection, they have different padding left. Numbered list has `padding-left: 30px` and the bulleted list has `padding-left: 20px`. Let's make them both 20px.**. This is happening in both the front end and the rich text editor.
>       Current:
>       Screenshot 2024-08-22 at 3 00 39 PM
>       Fixed:
>       Screenshot 2024-08-22 at 3 00 29 PM
> 
>     * [ ]  Not sure if there's a way to fix this, but I think it's confusing that the options are multi-select, and the class names are appended. Ideally, this should be a single option, and only one class should be applied at a time.
>       Screenshot 2024-08-22 at 2 57 42 PM
>       Screenshot 2024-08-22 at 2 57 53 PM
> 
>     * [x]  This can be a little nitpicky, but can we reorder the options so that the upperRoman is before the lower Roman, the same Upper Alpha before Lower Alpha so that are in the general order of hierarchy?

The 3rd point will be hard to achieve and it would require overriding the entire ckeditor build in drupal by adding a new plugin or utilizing other tools to make this happen.

**Review by clairesunstudio — APPROVED** [review]

**Pull request merged**